### PR TITLE
few things i've found out to be helpful for me

### PR DESCRIPTION
**Pull request by Need2Revolt**
I've found your software very helpful and i'm probably going to use it as a base for my satellite too, so i though i'd contribute a few things i've found along the way, of course feel free to discard everything if you don't like it

As the commit messages say:
- adding a more flexible sleep cycle management
i've found a bit complicated to have this split in several places and the reset procedure still remains untouched because of how the header files are built. I guess the final value for this will be dictated by the TTN fair use constraint, but for development it's useful to run a burst of measurements in a small time window

- logging frequency used for transmission
simple thing that was missing, it costed me some time, i hope it will save others

### CODE REVIEW

File: satellite-software/src/BME680Sensor.cpp [hunk 1]
```diff
@@ -303,8 +303,8 @@ const uint8_t* BME680Sensor::getCurrentMeasurementBuffer(void)
     PRINT_DEBUG(F("    calibrate temp = "));
     PRINT_DEBUG( temp_comp );
     PRINT_DEBUG(F(", normalize temp = "));
-    PRINT_DEBUG( (((float)temp_comp*9.0/5.0/100.0) + 32.0) );
-    PRINT_DEBUG(F(" °F, pressure = "));    
+    PRINT_DEBUG((float)temp_comp/100.0);
+    PRINT_DEBUG(F(" °C, pressure = "));
```
Comment (michaelkamprath) [inline satellite-software/src/BME680Sensor.cpp:307]: Sorry, I am not going to change the code from F to C. Can you make this a compile option instead? That way if the person wants the print out in F or C, they can chose. The default should be F because, well,  that's what I would make it. By making it a compile option, then someone can set the compile macro in the platformio.ini file, much like you did with the `CFG_eu868` compile option. Ideally things like this don't require users to change code to get optional behavior, they just change their compile configuration.

File: satellite-software/src/AmbaSat1App.cpp [hunk 2]
```diff
@@ -178,7 +182,7 @@ void AmbaSat1App::loop()
     // sleep device for designated sleep cycles
     for (int i=0; i < _config.getUplinkSleepCycles(); i++)
     {
-        LowPower.powerDown(SLEEP_8S, ADC_OFF, BOD_OFF);    //sleep 8 seconds * sleepcycles
+        LowPower.powerDown(TIME_INTERVAL, ADC_OFF, BOD_OFF);    //sleep TIME_INTERVAL seconds * sleepcycles
```
Comment (michaelkamprath) [inline satellite-software/src/AmbaSat1App.cpp:199]: You actually need use one of [the enums defined here](https://github.com/rocketscream/Low-Power/blob/master/LowPower.h#L6) for `TIME_INTERVAL`. You, of course, could define `TIME_INTERVAL` to be one of those enums, but you cannot define it as `30` as you do above. `LowPower.powerDown()` is a wrapper around the system level call `wdt_enable` ([see here for docs](https://www.nongnu.org/avr-libc/user-manual/group__avr__watchdog.html)), which also expects a finite set of possible values from a list of enumerations. Basically, the behavior of the code you implemented here is undefined because you are passing an invalid value for the first argument. The largest sleep interval the ATMEGA328P supports is 8 seconds. You just have to sequence multiple sleep intervals if you want to sleep longer.

With that said, the way I designed this code you shouldn't be changing the sleep interval size on this line. Trace back what `_config.getUplinkSleepCycles()` does. I intend for sleep interval to be a command-able configuration, meaning you send the device a command via TTN to change the sleep interval used. Given that, your change as-is is not something I would incorporate into the master code base. 

**EDIT:** Clearly I confused your `TIME_INTERVAL` with your `DEFAULT_RATE_VALUE`, but my general comment that changing the sleep rate is a run time configuration set through a command to the device still applies here. Changing the sleep rate should be transmitted to the device, not changed at compile time. Of course, it would be fair to add more flexibility what you can configure at run time. The relevant code that sets default values in in `PersistedConfiguration::resetToDefaults()` [here](https://github.com/michaelkamprath/ambasat1-flight-software/blob/master/satellite-software/src/PersistedConfiguration.cpp#L128).

**Review by michaelkamprath — CHANGES_REQUESTED** [review]
I'd be happy to incorporate the transmission frequency `PRINTLN_INFO` lines, but the other changes you made are more problematic. See my detailed comments.

File: satellite-software/src/BME680Sensor.cpp (hunk 1 above)
Comment (Need2Revolt) [reply satellite-software/src/BME680Sensor.cpp:307]: ah sorry, my bad, this was not supposed to be inside the pull request, it was meant for me for debugging, can you cherry pick the commit out of the pull request? or can i do it?

File: satellite-software/src/AmbaSat1App.cpp (hunk 2 above)
Comment (Need2Revolt) [reply satellite-software/src/AmbaSat1App.cpp:199]: yes, actually this: PersistedConfiguration::resetToDefaults() is the root cause of all this troubles.
I struggled a lot to find it and understand that is what i needed to modify to change the sleep cycle.
so what i tried to do is extrnalize it to a global definition that is easier to spot and use it in the constructor in a more explicit way (see     _config.setUplinkSleepCycles(DEFAULT_RATE_VALUE, true); in the setup() method).
other than that, i extrnalized the 8 seconds too, as they do work in pairs.

anyway, i realize this part of the change is messy and the default is not actually changed and i really couldn't do it, as per how the includes are organized.

so yeah, i understand if you want to reject this too

Comment (michaelkamprath) [reply satellite-software/src/AmbaSat1App.cpp:199]: I think you might be missing the design is doing. The function `PersistedConfiguration::resetToDefaults()` is ostensibly only called once ever across multiple reboots (reality, if you change the persisted memory bit layout, it gets called again). It saves the transmit rate to EEPROM on the ATMEGA328P. After that point, the code gets the transmit rate value from EEPROM, which you can change at run time. You should never have to change code in order to get a different transmit rate, you change the run time configuration value. to change that , you need to send a downlink to your device from the TTN web interface. The downlink is raw binary value, and I haven't yet documented it well, but you can [read more here](https://github.com/michaelkamprath/ambasat1-flight-software/tree/master/satellite-software#set-sensor-telemetry-uplink-rate). The whole idea here is so you can change the transmit rate of the satellite once it is in orbit, because once it is in orbit you cannot recompile the code. 

TL;DR: Changing the transmit rate using this code should not be done with compiler constants, but with satellite downlink commands.

File: satellite-software/platformio.ini [hunk 3]
```diff
@@ -9,7 +9,7 @@ lib_deps =
     https://github.com/matthijskooijman/arduino-lmic.git
 build_flags = 
     -D _lmic_config_h_
-    -D CFG_us915                ; change to CFG_eu868 in Europe
+    -D CFG_eu868
```
Comment (michaelkamprath) [inline satellite-software/platformio.ini:12]: I strongly suspect you are pushing you local build changes to this PR rather than the main branch in your forked repository. Changes like this I will not merge into my repository because as your commit title says, its your local build's config. 

Are you still intending for this PR to be merged into the main code?

File: satellite-software/platformio.ini [hunk 4]
```diff
@@ -25,8 +25,8 @@ build_flags =
     -D LOG_LEVEL=3
     -D LMIC_DEBUG_LEVEL=0
 ;	-D ENABLE_LOGGER_SENSOR
-monitor_port = /dev/cu.usbserial-14310
-upload_port = /dev/cu.usbserial-14310
+;monitor_port = COM6
+;upload_port = COM6
```
Comment (michaelkamprath) [inline satellite-software/platformio.ini:29]: As said above, this is really your local build config and should not be part of a PR.

**Review by michaelkamprath — CHANGES_REQUESTED** [review]

File: satellite-software/src/BME680Sensor.cpp (hunk 1 above)
Comment (michaelkamprath) [reply satellite-software/src/BME680Sensor.cpp:307]: I can't really cherry pick the pull request, and at this point, it would seem you have a lot of things in it that I won't merge because they pertain to your own personal configuration. Can you cancel this PR and try again? You should make a branch in your forked repository that represents what you are wanting to merge in with the main code.

**Comment by michaelkamprath** [comment]
@Need2Revolt I have reimplemented the changes I would merge from the PR with c0d98e1f91458fd5559589b0d0e73c3f2ecc661e. I am going to close this PR, but feel free to create a new PR for any new changes you might have. If you have further questions on how to use the satellite command code to do things like set the transmit frequency, please do not hesitate to open an issue.
